zebrad/src/components/mempool/tests/vector.rs

```diff
@@ -0,0 +1,482 @@
+
+    // Start the mempool service
+    let mut service = Mempool::new(
+        network,
+        Buffer::new(BoxService::new(peer_set), 1),
+        state_service.clone(),
+        tx_verifier,
+        sync_status,
+        latest_chain_tip,
+        chain_tip_change,
+    );
+
+    // Enable the mempool
+    let _ = service.enable(&mut recent_syncs).await;
+
+    // Insert the genesis block coinbase transaction into the mempool storage.
+    service.storage().insert(genesis_transaction.clone())?;
+
+    // Test `Request::TransactionIds`
+    let response = service
+        .ready_and()
+        .await
+        .unwrap()
+        .call(Request::TransactionIds)
+        .await
+        .unwrap();
+    let genesis_transaction_ids = match response {
+        Response::TransactionIds(ids) => ids,
+        _ => unreachable!("will never happen in this test"),
+    };
+
+    // Test `Request::TransactionsById`
+    let genesis_transactions_hash_set = genesis_transaction_ids
+        .iter()
+        .copied()
+        .collect::<HashSet<_>>();
+    let response = service
+        .ready_and()
+        .await
+        .unwrap()
+        .call(Request::TransactionsById(
+            genesis_transactions_hash_set.clone(),
+        ))
+        .await
+        .unwrap();
+    let transactions = match response {
+        Response::Transactions(transactions) => transactions,
+        _ => unreachable!("will never happen in this test"),
+    };
+
+    // Make sure the transaction from the blockchain test vector is the same as the
+    // response of `Request::TransactionsById`
+    assert_eq!(genesis_transaction, transactions[0]);
+
+    // Insert more transactions into the mempool storage.
+    // This will cause the genesis transaction to be moved into rejected.
+    // Skip the last (will be used later)
+    for tx in more_transactions {
+        service.storage().insert(tx.clone())?;
+    }
+
+    // Test `Request::RejectedTransactionIds`
+    let response = service
+        .ready_and()
+        .await
+        .unwrap()
+        .call(Request::RejectedTransactionIds(
+            genesis_transactions_hash_set,
+        ))
+        .await
+        .unwrap();
+    let rejected_ids = match response {
+        Response::RejectedTransactionIds(ids) => ids,
+        _ => unreachable!("will never happen in this test"),
+    };
+
+    assert_eq!(rejected_ids, genesis_transaction_ids);
+
+    // Test `Request::Queue`
+    // Use the ID of the last transaction in the list
+    let response = service
+        .ready_and()
+        .await
+        .unwrap()
+        .call(Request::Queue(vec![last_transaction.id.into()]))
+        .await
+        .unwrap();
+    let queued_responses = match response {
+        Response::Queued(queue_responses) => queue_responses,
+        _ => unreachable!("will never happen in this test"),
+    };
+    assert_eq!(queued_responses.len(), 1);
+    assert!(queued_responses[0].is_ok());
```

Comment:
No, that seems like some mistake I did while rebasing the changes. I checked for other things I might have missed, and there was a rename and declaration order change. I ended up changing the order but keeping the name because I thought it was a bit clearer :man_shrugging:

Reply:
No worries! I saw the other changes but they were harmless